File: STP/src/gestionAplicacion/Planeacion/Ruta.java
package gestionAplicacion.Planeacion;
import gestionAplicacion.Destinos.*;

import java.io.Serializable;
import java.util.ArrayList;

public class Ruta implements Serializable{
	private static final long serialVersionUID = 1L;
	private ArrayList<Conexion> ruta = new ArrayList<Conexion>();
	private double precio;
	private double distancia;
	private String tiempo;
	
	//constructores
	public Ruta(ArrayList<Conexion> ruta, double precio, double distancia, String tiempo) {
		super();
		this.ruta = ruta;
		this.precio = precio;
		this.distancia = distancia;
		this.tiempo = tiempo;
	}
	
	public Ruta() {
		super();
	}
	//getters and setters
	public ArrayList<Conexion> getRuta() {
		return ruta;
	}
	
	public void setRuta(ArrayList<Conexion> ruta) {
		this.ruta = ruta;
	}
	public double getPrecio() {
		return precio;
	}
	public void setPrecio(double precio) {
		this.precio = precio;
	}
	public double getDistancia() {
		return distancia;
	}
	public void setDistancia(double distancia) {
		this.distancia = distancia;
	}
	public String getTiempo() {
		return tiempo;
	}
	public void setTiempo(String tiempo) {
		this.tiempo = tiempo;
	}
	
	//metodos 

}

File: STP-Python/gestionAplicacion/Destinos/Ruta.py
from dis import dis


class Ruta():
    
    def __init__(self, *args):
        if len(args) == 4:
            self._ruta = args[0]
            self._precio = args[1]
            self._distanica = args[2]
            self._tiempo = args[3]
        elif len(args) == 0:
            self._ruta = []
            self._precio = []
            self._distanica = []
            self._tiempo = []
    
    def getRuta(self):
        return self._ruta

    def getPrecio(self):
        return self._precio

    def getDistancia(self):
        return self._distanica

    def getTiempo(self):
        return self._distanica

    def setRuta(self, ruta):
        self._ruta = ruta
    
    def setPrecio(self, precio):
        self._precio = precio
    
    def setDistancia(self, distancia):
        self._distanica = distancia
    
    def setTiempo(self, tiempo):
        self._tiempo = tiempo
    


